rpcserver.go

```diff
@@ -6745,6 +6751,222 @@ func (r *rpcServer) ListPermissions(_ context.Context,
+			ops.Actions = append(ops.Actions, fmt.Sprintf("%s added", addr.String()))
+			newAddrs = append(newAddrs, addr)
+		}
+	}
+
+	return newAddrs, ops, nil
+}
+
+// 	UpdateNodeAnnouncement allows the caller to update the node parameters
+//  and broadcasts a new version of the node announcement to its peers.
+func (r *rpcServer) UpdateNodeAnnouncement(_ context.Context,
+	req *lnrpc.NodeAnnouncementUpdateRequest) (
+	*lnrpc.NodeAnnouncementUpdateResponse, error) {
+
+	resp := &lnrpc.NodeAnnouncementUpdateResponse{}
+	nodeModifiers := make([]netann.NodeAnnModifier, 0)
+	currentNodeAnn, err := r.server.genNodeAnnouncement(false)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get current node announcement: %v", err)
+	}
+
+	if len(req.FeatureUpdates) > 0 {
+		ops, err := r.updateFeatures(req.FeatureUpdates)
+		if err != nil {
+			return nil, fmt.Errorf("error trying to update node features: %w", err)
+		}
+		resp.Ops = append(resp.Ops, ops)
+	}
+
+	if req.Color != "" {
+		color, err := parseHexColor(req.Color)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse color: %v", err)
+		}
+
+		if color != currentNodeAnn.RGBColor {
+			resp.Ops = append(resp.Ops, &lnrpc.Op{
+				Entity:  "color",
+				Actions: []string{fmt.Sprintf("changed to %v", color)}})
+			nodeModifiers = append(nodeModifiers, netann.NodeAnnSetColor(color))
+		}
+	}
+
+	if req.Alias != "" {
+		alias, err := lnwire.NewNodeAlias(req.Alias)
+		if err != nil {
+			return nil, fmt.Errorf("invalid alias value: %v", err)
+		}
+		if alias != currentNodeAnn.Alias {
+			resp.Ops = append(resp.Ops, &lnrpc.Op{
+				Entity:  "alias",
+				Actions: []string{fmt.Sprintf("changed to %v", alias)}})
+			nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAlias(alias))
+		}
+	}
+
+	if len(req.AddressUpdates) > 0 {
+		newAddrs, ops, err := r.updateAddresses(currentNodeAnn.Addresses, req.AddressUpdates)
+		if err != nil {
+			return nil, fmt.Errorf("error trying to update node addresses: %w", err)
+		}
+		resp.Ops = append(resp.Ops, ops)
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAddrs(newAddrs))
+	}
+
+	if len(nodeModifiers) == 0 {
+		return nil, fmt.Errorf("unable detect any new values to update " +
+			"the node announcement")
+	}
+
+	// Then, we'll generate a new timestamped node
+	// announcement with the updated addresses and broadcast
+	// it to our peers.
+	newNodeAnn, err := r.server.genNodeAnnouncement(
+		true, nodeModifiers...,
+	)
+	if err != nil {
+		rpcsLog.Debugf("Unable to generate new node "+
+			"node announcement: %v", err)
+		return nil, err
+	}
+
+	// Update the on-disk version of our announcement so
+	// other sub-systems will become out of sync
+	selfNode := &channeldb.LightningNode{
+		HaveNodeAnnouncement: true,
+		LastUpdate:           time.Unix(int64(newNodeAnn.Timestamp), 0),
+		Addresses:            newNodeAnn.Addresses,
+		Alias:                newNodeAnn.Alias.String(),
+		Features: lnwire.NewFeatureVector(
```

Comment:
this is what causes me a bit of discomfort, the endpoint will broadcast the node announcement changes but it can also be used for updating other feature sets?

Reply:
What do you mean other feature sets? You mean things that we don't actually understand?